nexus/db-queries/src/db/datastore/update.rs

```diff
@@ -206,6 +203,58 @@ impl DataStore {
             .map_err(|e| public_error_from_diesel(e, ErrorHandler::Server))
     }
 
+    /// List all TUF repositories (without artifacts) ordered by system version (newest first by default).
+    pub async fn tuf_repo_list_no_artifacts(
+        &self,
+        opctx: &OpContext,
+        pagparams: &DataPageParams<'_, Version>,
+    ) -> ListResultVec<TufRepo> {
+        opctx.authorize(authz::Action::Read, &authz::FLEET).await?;
+
+        use nexus_db_schema::schema::tuf_repo;
+
+        let conn = self.pool_connection_authorized(opctx).await?;
+
+        let marker_owner = pagparams
+            .marker
+            .map(|version| SemverVersion::from(version.clone()));
+        let db_pagparams = DataPageParams {
+            marker: marker_owner.as_ref(),
+            direction: pagparams.direction,
+            limit: pagparams.limit,
+        };
+
+        paginated(tuf_repo::table, tuf_repo::system_version, &db_pagparams)
+            .select(TufRepo::as_select())
+            .load_async(&*conn)
+            .await
+            .map_err(|e| public_error_from_diesel(e, ErrorHandler::Server))
+    }
+
+    /// List artifacts for a specific TUF repository by system version.
+    pub async fn tuf_repo_artifacts_list_by_version(
```

Comment:
only used in the integration test, since we no longer have the artifacts when we fetch a repo

Reply:
I'm wondering if we can make this a bit less of a footgun to have this here.

- It'd be nice if there were a way to actually restrict visibility of this but I don't know that there is.  Some of our crates using a `testing` feature?
- Failing that, can we name it something that indicates to somebody not to use it in real code?
- Can we add a `opctx.check_complex_operations_allowed()?`?  We might even be able to put this into `artifacts_for_repo` at this point, if there are no API endpoints that use that now.